pkg/sql/table_test.go

```diff
@@ -309,3 +314,33 @@ func TestPrimaryKeyUnspecified(t *testing.T) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 }
+
+func TestCanCloneTableWithUDT(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	ctx := context.Background()
+	params, _ := tests.CreateTestServerParams()
+	s, sqlDB, kvDB := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(ctx)
+	if _, err := sqlDB.Exec(`
+SET experimental_enable_enums=true;
+CREATE DATABASE test;
+CREATE TYPE test.t AS ENUM ('hello');
+CREATE TABLE test.tt (x test.t);
+`); err != nil {
+		t.Fatal(err)
+	}
+	desc := sqlbase.GetTableDescriptor(kvDB, keys.SystemSQLCodec, "test", "tt")
+	typLookup := func(id sqlbase.ID) (*tree.TypeName, *sqlbase.TypeDescriptor, error) {
+		typDesc, err := sqlbase.GetTypeDescFromID(ctx, kvDB, keys.SystemSQLCodec, id)
+		if err != nil {
+			return nil, nil, err
+		}
+		return &tree.TypeName{}, typDesc, nil
+	}
+	if err := sqlbase.HydrateTypesInTableDescriptor(desc, typLookup); err != nil {
+		t.Fatal(err)
+	}
+	// Ensure that we can clone this table.
+	_ = protoutil.Clone(desc).(*TableDescriptor)
```

Comment:
is there anything we can assert here? require.Equal(t, desc, protoutil.Clone(desc).(*TableDescriptor)) maybe?

Reply:
It just panics normally, so just being able to do it is what I want. I guess I could zero out the type metadata and check equality.